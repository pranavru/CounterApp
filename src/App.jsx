/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import ButtonComponent from './ButtonComponent';

class App extends Component {
  constructor() {
    super();
    this.state = {
      count: 0,
    };
  }

  makeIncrementer = (amount) => () => this.setState((prevState) => ({
    count: prevState.count + amount,
  }));

  makeDecrementer = (amount) => () => this.setState((prevState) => ({
    count: prevState.count - amount,
  }));

  render() {
    return (
      <div data-test="component-app">
        <h1>Learn React</h1>
        <p>
          Count:
          {this.state.count}
        </p>
        <ButtonComponent type="button" className="increment" clickedMethod={this.makeIncrementer(1)} name="Increment" />
        <ButtonComponent type="button" className="decrement" clickedMethod={this.makeDecrementer(1)} name="Decrement" />
      </div>
    );
  }
}

export default App;
