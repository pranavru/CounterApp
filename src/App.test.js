/* eslint-disable react/jsx-filename-extension */
// src/App.test.js

import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import App from './App';

Enzyme.configure({ adapter: new Adapter() });
describe('App Component', () => {
  // Using React Renderer
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders test with shallow & data-test without crashing', () => {
    const wrapper = shallow(<App />);
    const componentApp = wrapper.find("[data-test='component-app']");
    componentApp.debug();
    expect(componentApp.length).toBe(1);
  });

  test('should add', () => {
    expect(2 + 2).toBe(4);
  });

  test('should initialize the counter to zero', () => {
    const wrapper = shallow(<App />);
    const textField = wrapper.find('p').text();
    expect(textField).toBe('Count:0');
  });

  // Using Enzyme shallow
  test('should perform User Interaction to update Count by 1', () => {
    const wrapper = shallow(<App />);
    const incrementBtn = wrapper.find('ButtonComponent.increment');
    incrementBtn.simulate('click');
    const text = wrapper.find('p').text();
    expect(text).toEqual('Count:1');
  });

  test('decrements count by 1 when the decrement button is clicked', () => {
    const wrapper = shallow(<App />);
    const decrementBtn = wrapper.find('ButtonComponent.decrement');
    decrementBtn.simulate('click');
    const text = wrapper.find('p').text();
    expect(text).toEqual('Count:-1');
  });

  test('create a snapshot', () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// .mockImplementation((num) => (wrapper.setState({ count: wrapper.state('count') + (num * 10) })))
