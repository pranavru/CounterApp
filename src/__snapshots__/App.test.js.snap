// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component create a snapshot 1`] = `
<div
  data-test="component-app"
>
  <h1>
    Learn React
  </h1>
  <p>
    Count:
    0
  </p>
  <button
    className=""
    onClick={[Function]}
    type="button"
  >
    Increment
  </button>
  <button
    className=""
    onClick={[Function]}
    type="button"
  >
    Decrement
  </button>
</div>
`;
